Multiple secret sources docs (#8362)
solo-io/gloo: docs/content/guides/integrations/cert_manager/_index.md

   % curl https://$(glooctl proxy address | cut -f 1 -d ':').nip.io/api/pets -k
   [{"id":1,"name":"Dog","status":"available"},{"id":2,"name":"Cat","status":"pending"}]
   ```
5. Inspect the certificate that Envoy gets for this route.
   ```shell
   % openssl s_client -connect $(glooctl proxy address | cut -f 1 -d ':').nip.io:443
   ```

   Example output: You get back information about the certificate used for this connection, such as the following.

   ```
   subject=/CN=34.71.xx.xx.nip.io
   issuer=/CN=Fake LE Intermediate X1
   ```

You just confirmed that the service is accessible over the HTTPS port and that Envoy gets the certificate from Let's Encrypt!

## Use HashiCorp Vault as a Certificate Authority {#vault-ca}

Cert-manager supports using HashiCorp Vault as a CA. For Gloo Edge to use the certificates from Vault, you must set up a Vault instance as the CA. Then, set up cert-manager to use that Vault instance as an `Issuer`.

1. [Set up Vault as a CA by using the PKI secrets engine to generate certificates](https://developer.hashicorp.com/vault/docs/secrets/pki).
2. [Create a cert-manager `Issuer` for the Vault CA](https://cert-manager.io/docs/configuration/vault/).

> 3. If you use Vault to store other, non-TLS secrets along with the TLS certificates, then configure you default Gloo Edge Settings.

Author: The TLS secrets are stored in Kubernetes for this configuration
```suggestion
3. If you use Vault to store other, non-TLS secrets, then configure your default Gloo Edge Settings.
```

   ```shell
   kubectl -n gloo-system edit settings default
   ```
4. Update the Settings as follows:
   * Remove the existing `kubernetesSecretSource`, `vaultSecretSource`, or `directorySecretSource` field.
   * Add the `secretOptions` section with a Kubernetes source and a Vault source specified to enable secrets to be read from both Kubernetes and Vault.
   * Add the `refreshRate` field to watch for changes in Vault secrets and the local filesystem of where Gloo Edge runs.
   {{< highlight yaml "hl_lines=16-28" >}}
   apiVersion: gloo.solo.io/v1
   kind: Settings
   metadata:
     name: default
     namespace: gloo-system
   spec:
     discoveryNamespace: gloo-system
     gateway:
       validation:
         alwaysAccept: true
         proxyValidationServerAddr: gloo:9988
     gloo:
       xdsBindAddr: 0.0.0.0:9977
     kubernetesArtifactSource: {}
     kubernetesConfigSource: {}

>      # Delete or comment out the existing \*SecretSource field

Author: this didn't render correctly. going to check locally before committing these changes

```suggestion
     # Delete or comment out the existing *SecretSource field
```
